fdbserver/storageserver.actor.cpp

```diff
@@ -9135,9 +9214,42 @@ ACTOR Future<Void> fetchKeys(StorageServer* data, AddingShard* shard) {
 				TraceEvent(bulkLoadVerboseEventSev(), "SSBulkLoadTaskFetchKey", data->thisServerID)
 				    .detail("DataMoveId", dataMoveId.toString())
 				    .detail("Range", keys)
+				    .detail("Knobs", SERVER_KNOBS->BULK_LOAD_USE_SST_INGEST)
+				    .detail("SupportsSstIngestion", data->storage.getKeyValueStore()->supportsSstIngestion())
 				    .detail("Phase", "File download");
-				hold = tryGetRangeForBulkLoad(results, keys, localBulkLoadFileSets);
-				rangeEnd = keys.end;
+				// Attempt SST ingestion...
+				if (SERVER_KNOBS->BULK_LOAD_USE_SST_INGEST &&
+				    data->storage.getKeyValueStore()->supportsSstIngestion()) {
+					TraceEvent(bulkLoadVerboseEventSev(), "SSBulkLoadTaskFetchKey", data->thisServerID)
+					    .detail("DataMoveId", dataMoveId.toString())
+					    .detail("Range", keys)
+					    .detail("Phase", "SST ingestion");
+					// Verify ranges...
+					for (const auto& [range, fileSet] : *localBulkLoadFileSets) {
+						ASSERT(keys.contains(range));
+					}
+					// Clear the key range before ingestion. This mirrors the replaceRange done in the case were
+					// we do not ingest SST files.
+					data->storage.getKeyValueStore()->clear(keys);
+					// Compact the range before ingestion to optimize storage
+					wait(data->storage.getKeyValueStore()->compactRange(keys));
```

Comment:
We may want to move this compactRange after the wait(data->durableVersion.whenAtLeast(data->storageVersion() + 1));

Reply:
Done (The argument is that clear has not landed in rocksdb, not until we've seen version increment.... Running a compaction before version change may have us compacting before the clear range hits rocksdb)